bigquery/google/cloud/bigquery/job.py

```diff
@@ -2846,6 +2846,36 @@ def _blocking_poll(self, timeout=None):
         self._done_timeout = timeout
         super(QueryJob, self)._blocking_poll(timeout=timeout)
 
+    @staticmethod
+    def _format_for_exception(query, job_id):
+        """Format a query for the output in exception message.
+
+        Args:
+            query (str): The SQL query to format.
+            job_id (str): The ID of the job that ran the query.
+
+        Returns: (str)
+            A formatted query text.
+        """
+        template = "\n\n(job ID: {job_id})\n\n{header}\n\n{ruler}\n{body}\n{ruler}"
+
+        lines = query.splitlines()
+        max_line_len = max(len(l) for l in lines)
+
+        header = "-----Offending SQL Follows-----"
```

Comment:
An error with the query isn't always the SQL's fault. Sometimes there is just resource contention or some other transient failure.

I recommend changing the header to `Query Job SQL` or something along those lines.

Reply:
A good point indeed, will generalize the header. :+1: